README.md

```diff
@@ -6,41 +6,126 @@ Transform FHIR to dataset for ML applications
-The purpose of this file is to perform a query whose result will be identical to the next SQL query:
+## Installation
+
+After cloning this repository, you can install the required dependencies
+
+```
+pip install -r requirements.txt
+npm install --prefix ./fhir2dataset/metadata
+```
+
+For usage, refer to this [turorial](https://htmlpreview.github.io/?https://github.com/arkhn/FHIR2Dataset/blob/query_tests/examples/tutorial.html) and then this [Jupyer Notebook](examples/example.ipynb)
+
+
+## Getting started
+
+Two possible ways to enter the query : as a SQL query or as a JSON config file
+
+**SQL query as entry**
+
+```
+import pandas as pd
+from fhir2dataset.query import Query
+from fhir2dataset.fhirrules_getter import FHIRRules
+from fhir2dataset.parser import FHIR2DatasetParser
+
+fhir_api_url = 'http://hapi.fhir.org/baseR4/'
+fhir_rules=FHIRRules(fhir_api_url=fhir_api_url)
+query = Query(fhir_api_url,fhir_rules=fhir_rules)
+parser = FHIR2DatasetParser()
+```
 
 ```
-SELECT (alias n°1).a, (alias n°1).b, (alias n°1).c, (alias n°2).a FROM (Resource type 1) as (alias n°1)
+sql_like_query = "SELECT (alias n°1).a, (alias n°1).b, (alias n°1).c, (alias n°2).a FROM (Resource type 1) as (alias n°1)
 INNER JOIN (Resource type 2) as (alias n°2)
 ON (alias n°1).d = (alias n°2)
 INNER JOIN (Resource type 3) as (alias n°3)
 ON (alias n°2).b = (alias n°3) WHERE (alias n°2).c = "value 1"
 AND (alias n°2).d = "value 2"
 AND (alias n°3).a = "value 3"
-AND (alias n°3).b = "value 4"
+AND (alias n°3).b = "value 4""
 ```
 
-## Installation
+```
+config_from_parser = parser.parse(sql_like_query)
+query.from_config(config_from_parser)
+query.execute()
+``` 
 
-After cloning this repository, you can install the required dependencies
+**JSON config file as entry**
 
 ```
-pip install -r requirements.txt
-npm install --prefix ./fhir2dataset/metadata
+import pandas as pd
+from fhir2dataset.query import Query
+from fhir2dataset.fhirrules_getter import FHIRRules
+
+fhir_api_url = 'http://hapi.fhir.org/baseR4/'
+fhir_rules=FHIRRules(fhir_api_url=fhir_api_url)
+query = Query(fhir_api_url,fhir_rules=fhir_rules)
+```
+
+config.json :
+
+```json
+{"select":{
+    "alias n°1":[
+        "a",
+        "b",
+        "c"
+    ],
+    "alias n°2":[
+        "a"
+    ]
+},
+"from":{
+    "alias n°1":"Resource type 1",
+    "alias n°2":"Resource type 2",
+    "alias n°3":"Resource type 3"
+},
+"join":{
+    "inner": {
+        "alias n°1":{
+            "d":"alias n°2"
+
+        },
+        "alias n°2":{
+            "b":"alias n°3"
+        }
+
+    }
+},
+"where":{
+    "alias n°2":{
+        "c":"value 1",
+        "d":"value 2"
+    },
+    "alias n°3":{
+        "a":"value 3",
+        "b":"value 4"
+    }
+}}
+```
+```
+# Enter in dirname the path of config.json
+filename_config = 'config.json'
+
+with open(os.path.join(dirname, filename_config)) as json_file:
+    config = json.load(json_file)
+
+query.from_config(config)
+query.execute()
```

Comment:
Maybe we could put forward the recovery of the table resulting from the query, what do you think?

```suggestion
query.execute()
df = query.main_dataframe
```

Reply:
Yes it is much better !